hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/TokenFeature.java

```diff
@@ -421,6 +528,77 @@ public void verifyMirrorTokenRelationshipNftAPIResponses() {
         assertThat(token.getKycStatus()).isEqualTo(MirrorKycStatus.NOT_APPLICABLE);
     }
 
+    @Given("^I approve sender(?: (.*))? to transfer up to (.*) tokens (?:(.*))?$")
+    public void setFungibleTokenAllowance(Integer senderIndex, long amount, Integer tokenIndex) {
+        var tokenId = tokenIds.get(getIndexOrDefault(tokenIndex));
+        var spenderAccountId = senders.get(getIndexOrDefault(senderIndex));
+        networkTransactionResponse = accountClient.approveToken(tokenId, spenderAccountId.getAccountId(), amount);
+        assertNotNull(networkTransactionResponse.getTransactionId());
+        assertNotNull(networkTransactionResponse.getReceipt());
+    }
+
+    @Given("I delete the allowance on token {int} for sender {int}")
+    public void setFungibleTokenAllowance(Integer tokenIndex, Integer senderIndex) {
+        var tokenId = tokenIds.get(getIndexOrDefault(tokenIndex));
+        var spenderAccountId = senders.get(getIndexOrDefault(senderIndex));
+        networkTransactionResponse = accountClient.approveToken(tokenId, spenderAccountId.getAccountId(), 0L);
+        assertNotNull(networkTransactionResponse.getTransactionId());
+        assertNotNull(networkTransactionResponse.getReceipt());
+    }
+
+    @Then(
+            "the mirror node REST API should confirm the approved token (?:(.*) )?allowance of (.*) for sender(?: (.*))?$")
+    public void verifyMirrorAPIApprovedTokenAllowanceResponse(
```

Comment:
add the annotation to retry all these public methods which assert certain value from mirrornode rest api since it takes time for the changes being finalized in consensus nodes then propagated and persisted by mirrornode importer

also check the crypto allowance changes make sure it's not missed there.

```suggestion
    @RetryAsserts
    public void verifyMirrorAPIApprovedTokenAllowanceResponse(
```